packages/database/src/migrations/20210912223650-LESMISVitalsUseEntityAggregation-modifies-data.js

```diff
@@ -0,0 +1,232 @@
+'use strict';
+
+import { insertObject, deleteObject, generateId, findSingleRecord } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const DEVELOPMENT_PARTNERS = [
+  'SchDP_AEAL',
+  'SchDP_CRS',
+  'SchDP_HII',
+  'SchDP_Plan',
+  'SchDP_RtR',
+  'SchDP_UNICEF',
+  'SchDP_WB',
+  'SchDP_WC',
+  'SchDP_WFP',
+  'SchDP_WR',
+  'SchDP_WV',
+  'SchCVD023',
+];
+
+const OBSOLETE_REPORTS = [
+  'LESMIS_village_vitals',
+  'LESMIS_sub_district_vitals',
+  'LESMIS_multi_school_vitals',
+];
+
+// New indicators
+const generateDevelopmentPartnerIndicator = baseDataElement => ({
+  id: generateId(),
+  code: `${baseDataElement}_summary`,
+  builder: 'analyticArithmetic',
+  config: {
+    formula: `${baseDataElement} > 0`,
+    aggregation: [
+      'MOST_RECENT',
+      {
+        type: 'COUNT_PER_ORG_GROUP',
+        config: {
+          dataSourceEntityType: 'school',
+          aggregationEntityType: 'requested',
+        },
+      },
+    ],
+    defaultValues: {
+      [`${baseDataElement}`]: 0,
+    },
+  },
+});
+
+// ece, lse, use already exist from WAI-44
+const NOSTU_PRIMARY_EDUCATION_TOTAL_INDICATOR = {
+  id: generateId(),
+  code: 'nostu_pe_t',
+  builder: 'analyticArithmetic',
+  config: {
+    formula: 'nostu_pe_f_public + nostu_pe_f_private + nostu_pe_m_public + nostu_pe_m_private',
+    aggregation: [
+      'MOST_RECENT',
+      {
+        type: 'SUM_PER_ORG_GROUP',
+        config: {
+          dataSourceEntityType: 'sub_district',
+          aggregationEntityType: 'requested',
+        },
+      },
+    ],
+    defaultValues: {
+      nostu_pe_f_public: 0,
+      nostu_pe_f_private: 0,
+      nostu_pe_m_public: 0,
+      nostu_pe_m_private: 0,
+    },
+  },
+};
+
+const NOSTU_TOTAL = {
+  id: generateId(),
+  code: 'nostu_total',
+  builder: 'analyticArithmetic',
+  config: {
+    formula: 'nostu_ece_t + nostu_pe_t + nostu_lse_t + nostu_use_t',
+    aggregation: [
+      'MOST_RECENT',
+      {
+        type: 'SUM_PER_ORG_GROUP',
+        config: {
+          dataSourceEntityType: 'sub_district',
+          aggregationEntityType: 'requested',
+        },
+      },
+    ],
+    defaultValues: {
+      nostu_ece_t: 0,
+      nostu_pe_t: 0,
+      nostu_lse_t: 0,
+      nostu_use_t: 0,
+    },
+  },
+};
+
+// New reports
+const ENTITY_VITALS_REPORT_CONFIG = {
+  fetch: {
+    dataElements: [
+      'SDP001',
+      'nosch_total',
+      'nostu_total',
+      ...DEVELOPMENT_PARTNERS.map(dataElement => `${dataElement}_summary`),
```

Comment:
With the new indicators all of these can be requested from the `sub_district` level and aggregated up, this means we can use one report for all entity levels (except school which requires specific values)